File: 백준/Silver/1920. 수 찾기/수 찾기.py
import sys

N = int(sys.stdin.readline().strip())
numN = list(map(int, sys.stdin.readline().strip().split(' ')))
M = int(sys.stdin.readline().strip())
numM = list(map(int, sys.stdin.readline().strip().split(' ')))
numN.sort()

def search(arr, target):
  left = 0
  right = len(arr) - 1
  # print('arr',arr,'target',target)
  while left <= right:
    mid = (left + right) // 2
    if arr[mid] == target:
      return 1
    elif target <= arr[mid]:
      right = mid - 1
    else:
      left = mid + 1

  return 0

for m in numM:
  print(search(numN, m))

File: 백준/Silver/1920. 수 찾기/수 찾기.js
/* /dev/stdin */
let fs = require("fs");
let input = fs
  .readFileSync("/dev/stdin")
  .toString()
  .split("\n")
  .map((x) => x.replaceAll("\r", ""));

const N = Number(input[0]);
const A = input[1]
  .split(" ")
  .map(Number)
  .sort((a, b) => a - b);

const M = Number(input[2]);
const arrM = input[3].split(" ").map(Number);

let answer = [];

for (const x of arrM) {
  answer.push(binarySearch(A, x));
}

console.log(answer.join("\n"));

function binarySearch(arr, target) {
  let left = 0;
  let right = arr.length - 1;

  while (left <= right) {
    const mid = Math.floor((left + right) / 2);

    if (arr[mid] === target) {
      return 1;
    } else if (arr[mid] < target) {
      left = mid + 1;
    } else {
      right = mid - 1;
    }
  }

  return 0;
}
